presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -332,19 +337,37 @@ protected Type visitDereferenceExpression(DereferenceExpression node, StackableA
             return rowFieldType;
         }
 
-        private void assertColumnPrefix(QualifiedName qualifiedName, Expression node)
+        private boolean isColumnPrefix(QualifiedName qualifiedName, RelationType tupleDescriptor)
         {
             // Recursively check if its prefix is a column.
             while (qualifiedName.getPrefix().isPresent()) {
                 qualifiedName = qualifiedName.getPrefix().get();
                 List<Field> matches = tupleDescriptor.resolveFields(qualifiedName);
                 if (!matches.isEmpty()) {
-                    // The AMBIGUOUS_ATTRIBUTE exception will be thrown later with the right node if matches.size() > 1
-                    return;
+                    return true;
                 }
             }
+            return false;
+        }
 
-            throw createMissingAttributeException(node);
+        private boolean isReferenceToOuterRelation(AnalysisContext context, QualifiedName qualifiedName)
```

Comment:
This can be simplified slightly to:

```java
AnalysisContext current = context;
while (current != null) {
    RelationType type = current.getParentRelationType();
    if (!type.resolveFields(qualifiedName).isEmpty()) {
        return true;
    }
    if (isColumnPrefix(qualifiedName, type)) {
        return true;
    }
    current = current.getParent().orElse(null);
}
return false;
```